src/main/java/org/bonitasoft/connectors/ws/SecureWSConnector.java

```diff
@@ -244,6 +279,58 @@ protected void executeBusinessLogic() throws ConnectorException {
         setOutputParameter(OUTPUT_RESPONSE_DOCUMENT_BODY, responseDocumentBody);
     }
 
+    private void sanitizeInputs() {
+
+        Map<String, Object> sanitizedInputs = new HashMap<>();
+        sanitizedInputs.put(HTTP_HEADERS, sanitizeHTTPHeaders((List<List<Object>>) getInputParameter(HTTP_HEADERS)));
+        sanitizedInputs.put(PASSWORD, sanitizeString((String) getInputParameter(PASSWORD)));
+        sanitizedInputs.put(USER_NAME, sanitizeString((String) getInputParameter(USER_NAME)));
+        sanitizedInputs.put(BINDING, sanitizeString((String) getInputParameter(BINDING)));
+        sanitizedInputs.put(SOAP_ACTION, sanitizeString((String) getInputParameter(SOAP_ACTION)));
+        sanitizedInputs.put(ENDPOINT_ADDRESS, sanitizeString((String) getInputParameter(ENDPOINT_ADDRESS)));
+        sanitizedInputs.put(ENVELOPE, sanitizeString((String) getInputParameter(ENVELOPE)));
+        sanitizedInputs.put(PORT_NAME, sanitizeString((String) getInputParameter(PORT_NAME)));
+        sanitizedInputs.put(SERVICE_NAME, sanitizeString((String) getInputParameter(SERVICE_NAME)));
+        sanitizedInputs.put(SERVICE_NS, sanitizeString((String) getInputParameter(SERVICE_NS)));
+        sanitizedInputs.put(PROXY_HOST, sanitizeString((String) getInputParameter(PROXY_HOST)));
+        sanitizedInputs.put(PROXY_PORT, sanitizeString((String) getInputParameter(PROXY_PORT)));
+        sanitizedInputs.put(PROXY_PROTOCOL, sanitizeString((String) getInputParameter(PROXY_PROTOCOL)));
+        sanitizedInputs.put(PROXY_USER, sanitizeString((String) getInputParameter(PROXY_USER)));
+        sanitizedInputs.put(PROXY_PASSWORD, sanitizeString((String) getInputParameter(PROXY_PASSWORD)));
+        //Booleans, no need to sanitize them
+        sanitizedInputs.put(PRINT_REQUEST_AND_RESPONSE, getInputParameter(PRINT_REQUEST_AND_RESPONSE));
+        sanitizedInputs.put(BUILD_RESPONSE_DOCUMENT_BODY, getInputParameter(BUILD_RESPONSE_DOCUMENT_BODY));
+        sanitizedInputs.put(BUILD_RESPONSE_DOCUMENT_ENVELOPE, getInputParameter(BUILD_RESPONSE_DOCUMENT_ENVELOPE));
+        // Replace Inputs with sanitized version
+        setInputParameters(sanitizedInputs);
+    }
+
+    private List<List<Object>> sanitizeHTTPHeaders(List<List<Object>> httpHeadersList) {
```

Comment:
Is it necessary to sanitized the http headers? I though that only the body was converted into XML

Reply:
Most important is the ENVELOPE